includes/Classifai/Features/ExcerptGeneration.php

```diff
@@ -184,13 +190,44 @@ public function rest_endpoint_callback( WP_REST_Request $request ) {
 		$route = $request->get_route();
 
 		if ( strpos( $route, '/classifai/v1/generate-excerpt' ) === 0 ) {
+			$post_id = $request->get_param( 'id' );
+
+			// Get the author name - prefer from request payload, fallback to database.
+			$author_name = $request->get_param( 'author' );
+
+			if ( empty( $author_name ) && $post_id ) {
+				$post = get_post( $post_id );
+				if ( $post ) {
+					$author_name = get_the_author_meta( 'display_name', $post->post_author );
+				}
+			}
+
+			if ( $post_id ) {
+				$post = get_post( $post_id );
+
+				/**
+				 * Filter the author name used in excerpt generation.
+				 *
+				 * @since 3.x.x
+				 * @hook classifai_excerpt_generation_author_name
+				 *
+				 * @param {string} $author_name The author's display name.
+				 * @param {int}    $post_id     The post ID.
+				 * @param {WP_Post} $post       The post object.
+				 *
+				 * @return {string} The author name to use in the prompt.
+				 */
+				$author_name = apply_filters( 'classifai_excerpt_generation_author_name', $author_name, $post_id, $post );
+			}
```

Comment:
I think we would want this filter to always run, even if we don't have a post ID. So I'd remove that conditional. In addition, I don't think we need to pass the post object to the filter as someone can get that themselves if they need it.
```suggestion
			/**
			 * Filter the author name used in excerpt generation.
			 *
			 * @since x.x.x
			 * @hook classifai_excerpt_generation_author_name
			 *
			 * @param {string} $author_name The author's display name.
			 * @param {int}    $post_id     The post ID.
			 *
			 * @return {string} The author name to use in the prompt.
			 */
			$author_name = apply_filters( 'classifai_excerpt_generation_author_name', $author_name, $post_id );
```